### ISSUE TITLE
Publishing to a Windows Virtual Machine on Azure Topic

### ISSUE THREAD

**Issue by GuardRex** [aspnet/Docs#256]
@danroth27 @Rick-Anderson Hey guys, has http://docs.asp.net/en/latest/publishing/azure-windows-vm.html been assigned? ... If not, may I take a stab at it?

**Comment by Rick-Anderson** [comment]
Doesn’t look like it. Go for it.

**Comment by GuardRex** [comment]
Thanks ... I'll have a draft PR tomorrow that we can take a look at.

**Comment by danroth27** [comment]
@GuardRex Thanks for offering to contribute! Before you do the work to submit a full PR it would be good to start off with an outline proposal. That way we can make sure we are on the same page on what content the article will contain.

**Comment by GuardRex** [comment]
@danroth27 Sure thing ... rattling the key elements off the top of my head:

1. Brief abstract describing what will be covered
2. Bulleted list of topics
3. Topic 1: Local Environment Configuration (most notably installing Web Deploy)
4. Topic 2: Project Configuration (most notably the dependency on `Microsoft.AspNet.Server.IIS`)
5. Topic 3: Server/VM Configuration (most notably the presence of .NET Framework 4.5.1, Web Deploy, project folder layout, IIS website & AppPool settings)
6. Topic 4: MSDeploy Commands: Stop the AppPool (to release IIS file locks), deploy the application, and restart the AppPool
7. Summary
8. Additional Resources

I'll add the "gotchas" that I ran into and reported in https://github.com/aspnet/Hosting/issues/141 and https://github.com/aspnet/Home/issues/452 along the way, and I'd like to have good screenshot coverage that makes it very clear what needs to be done. I'd write from the perspective that someone has never used IIS Manager before but does have a running Azure VM.

The only thing that I don't know yet is if the three MSDeploy commands can be scripted and triggered by `postpublish`. I'll play with that and see what happens. I've been issuing the commands manually, but I want them scripted, so that's the next logical step in my investigation. If that turns out that scripting the commands will work, then scripting would be added to Topic 2: Project Configuration and Topic 4: MSDeploy Commands would be set into Topic 1: Local Environment Configuration, where I would talk about how to script the commands. I think it would be ok to structure things this way because I would have already laid out the content of the topics in the abstract ... but feel free to direct me on what you want.

... and on a broader note, if I'm way off base here, just give me the word. Writing this would be a natural offshoot of the issues I was discussing with the team on this subject and allow me to contribute along the lines of my strengths (I was a college prof for years); however, I would certainly step aside and tip my hat to someone more qualified/knowledgeable for the job.

**Comment by GuardRex** [comment]
... oh ... and one more bit ... I'll add in a blurb on the Azure endpoint port mapping for the Web Deployment Agent Service (to port 8172) and VM firewall configuration to allow traffic on 8172 and link to those MSDN docs. I think it would be best if the docs don't assume that someone has any of this knowledge and needs links to walk-throughs on those topics.

**Comment by danroth27** [comment]
Thanks! I'm working on trying to find someone with the right context to help review this content.

**Comment by GuardRex** [comment]
@danroth27 @Rick-Anderson I have an update:

Yes! The MSDeploy commands can be put into a batch file and triggered to run after a publish to the file system.

However, I'm checking with the team to see about the behavior of the `postpublish` script execution setting in `project.json` ... It's executing the script **before** it actually publishes the app to the `bin/output` folder ... no good for our purposes here. I simply worked around it by triggering the batch file to run in the VS Publish PowerShell script. I posed my ? to them here: https://github.com/aspnet/Home/issues/622

I won't work on docs until I hear back from you, but I think I have a good recipe to tide folks over with Web Deploy until the tooling cats make it sweet in VS.
